tests/functional/DatadirTest.php

```diff
@@ -0,0 +1,437 @@
+                                ],
+                            ],
+                        ],
+                    ],
+                ],
+            ],
+        ];
+        // phpcs:enable
+
+        $this->runAppWithConfig($configArray);
+
+        $config = $this->getConfigFromUserConfig($configArray);
+
+        $connection = $this->getConnection($config);
+
+        $insertedData = $connection->fetchAll(
+            sprintf('SELECT * FROM %s', QueryBuilder::quoteIdentifier('query_tag'))
+        );
+
+        $this->assertNotEmpty($insertedData);
+        $expectedData = sprintf('{"runId":"%s"}', getenv('KBC_RUNID'));
+        $this->assertEquals($expectedData, $insertedData[0]['QUERY_TAG']);
+    }
+
+    public function testAbortTransformation(): void
+    {
+        $config = [
+            'authorization' => $this->getDatabaseConfig(),
+            'parameters' => [
+                'blocks' => [
+                    [
+                        'name' => 'first block',
+                        'codes' => [
+                            [
+                                'name' => 'first code',
+                                'script' => [
+                                    'SET ABORT_TRANSFORMATION=\'Abort Me Please\'',
+                                ],
+                            ],
+                        ],
+                    ],
+                ],
+            ],
+        ];
+
+        $this->runAppWithConfig(
+            $config,
+            1,
+            null,
+            "Transformation aborted with message \"Abort Me Please\"\n"
+        );
+    }
+
+    private function runAppWithConfig(
+        array $config,
+        int $expectedReturnCode = 0,
+        ?string $expectedStdout = null,
+        ?string $expectedStderr = null
+    ): Process {
+        $specification = new DatadirTestSpecification(
+            null,
+            $expectedReturnCode,
+            $expectedStdout,
+            $expectedStderr,
+            null
+        );
+
+        $tempDatadir = $this->getTempDatadir($specification);
+
+        file_put_contents($tempDatadir->getTmpFolder() . '/config.json', json_encode($config));
+
+        $process = $this->runScript($tempDatadir->getTmpFolder());
+
+        $this->assertMatchesSpecification($specification, $process, $tempDatadir->getTmpFolder());
+
+        return $process;
+    }
+
+    private function getConfigFromUserConfig(array $userConfig): Config
+    {
+        return new Config($userConfig, new ConfigDefinition());
+    }
+
+    private function getConnection(Config $config): Connection
+    {
+        return new Connection($config->getDatabaseConfig());
+    }
+
+    private function getDatabaseConfig(): array
+    {
+        return [
+            'workspace' => [
+                'host' => getenv('SNOWFLAKE_HOST'),
+                'port' => getenv('SNOWFLAKE_PORT'),
```

Comment:
vcera jsem prisel na to, ze ten port tam nebude, connection ho nevraci a vzdycky to je na defaultu

Reply:
ok, dal jsem ho pryč